rustls/src/webpki/verify.rs

```diff
@@ -65,6 +66,38 @@ pub fn verify_server_cert_signed_by_trust_anchor(
         .map(|_| ())
 }
 
+/// Verify that the end-entity certificate `end_entity` is a valid client cert
+/// and chains to at least one of the [OwnedTrustAnchor] in the `roots` [RootCertStore].
+///
+/// `intermediates` contains all certificates other than `end_entity` that
+/// were sent as part of the client's [Certificate] message. It is in the
+/// same order that the client sent them and may be empty.
+#[allow(dead_code)]
+#[cfg_attr(not(feature = "dangerous_configuration"), allow(unreachable_pub))]
+pub fn verify_client_cert_signed_by_trust_anchor(
+    cert: &ParsedCertificate,
+    roots: &RootCertStore,
+    intermediates: &[Certificate],
+    now: SystemTime,
+    revocation: Option<RevocationOptions<'_>>,
+) -> Result<(), Error> {
+    let chain = intermediate_chain(intermediates);
+    let trust_roots = trust_roots(roots);
+    let webpki_now = webpki::Time::try_from(now).map_err(|_| Error::FailedToGetCurrentTime)?;
+
+    cert.0
+        .verify_for_usage(
+            SUPPORTED_SIG_ALGS,
+            &trust_roots,
+            &chain,
+            webpki_now,
+            webpki::KeyUsage::client_auth(),
+            revocation, // no CRLs
```

Comment:
The `no CRLs` comment doesn't make sense in the client context, where a `Option<RevocationOptions<'_>>` is actually passed in.

(Is `RevocationOptions<'_>` part of the public API?

Reply:
`RevocationOptions` is part of the webpki public API

Reply:
I prefer so, If user don't want to check CRLs, they can pass None

Reply:
We don't want to have webpki public API in the rustls public API, so we'll need a different solution here.

Reply:
`WebPkiClientVerifier#new` already rely on `webpki::OwnedCertRevocationList`